src/compositor/seat/pointer.c

```diff
@@ -225,7 +227,7 @@ focus_view(struct wlc_pointer *pointer, struct wlc_surface *surf, wlc_handle old
          wlc_log(WLC_LOG_WARN, "Failed to push focused pointer resource to pool (out of memory?)");
 
       uint32_t serial = wl_display_next_serial(wlc_display());
-      wl_pointer_send_enter(wr, serial, surface, wl_fixed_from_double(pos->x), wl_fixed_from_double(pos->y));
+      wl_pointer_send_enter(wr, serial, surface, wl_fixed_from_double(pos->x/scale), wl_fixed_from_double(pos->y/scale));
```

Comment:
I will skip the view.c and pointer.c code changes and detail my thoughts about how this functionality could be achieved with less changes in separate comment.